diff --git a/node_modules/react-native-android-location-enabler/android/build.gradle b/node_modules/react-native-android-location-enabler/android/build.gradle
index a74704f..ba94517 100644
--- a/node_modules/react-native-android-location-enabler/android/build.gradle
+++ b/node_modules/react-native-android-location-enabler/android/build.gradle
@@ -8,7 +8,7 @@ buildscript {
   }
 
   dependencies {
-    classpath "com.android.tools.build:gradle:7.2.2"
+    classpath "com.android.tools.build:gradle:8.9.1"
     // noinspection DifferentKotlinGradleVersion
     classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
   }
@@ -99,7 +99,7 @@ dependencies {
   implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
   implementation "com.google.android.gms:play-services-location:21.0.1"
   implementation 'androidx.appcompat:appcompat:1.6.1'
-  implementation platform('org.jetbrains.kotlin:kotlin-bom:1.8.0')
+  implementation platform('org.jetbrains.kotlin:kotlin-bom:2.1.20')
 }
 
 if (isNewArchitectureEnabled()) {
diff --git a/node_modules/react-native-android-location-enabler/android/gradle.properties b/node_modules/react-native-android-location-enabler/android/gradle.properties
index 50ee582..fab3001 100644
--- a/node_modules/react-native-android-location-enabler/android/gradle.properties
+++ b/node_modules/react-native-android-location-enabler/android/gradle.properties
@@ -1,5 +1,5 @@
-AndroidLocationEnabler_kotlinVersion=1.9.0
-AndroidLocationEnabler_minSdkVersion=21
-AndroidLocationEnabler_targetSdkVersion=31
-AndroidLocationEnabler_compileSdkVersion=31
+AndroidLocationEnabler_kotlinVersion=2.1.20
+AndroidLocationEnabler_minSdkVersion=24
+AndroidLocationEnabler_targetSdkVersion=35
+AndroidLocationEnabler_compileSdkVersion=35
 AndroidLocationEnabler_ndkversion=21.4.7075529
diff --git a/node_modules/react-native-android-location-enabler/android/src/main/java/com/heanoria/library/reactnative/locationenabler/AndroidLocationEnablerModule.kt b/node_modules/react-native-android-location-enabler/android/src/main/java/com/heanoria/library/reactnative/locationenabler/AndroidLocationEnablerModule.kt
index 00d8839..2d69a89 100644
--- a/node_modules/react-native-android-location-enabler/android/src/main/java/com/heanoria/library/reactnative/locationenabler/AndroidLocationEnablerModule.kt
+++ b/node_modules/react-native-android-location-enabler/android/src/main/java/com/heanoria/library/reactnative/locationenabler/AndroidLocationEnablerModule.kt
@@ -43,7 +43,7 @@ class AndroidLocationEnablerModule(reactContext: ReactApplicationContext) :
 
   @ReactMethod
   fun isLocationEnabled(promise: Promise) {
-    if (currentActivity == null) return
+    if (reactApplicationContext.currentActivity == null) return
 
     val locationManager = this.context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
 
@@ -54,7 +54,7 @@ class AndroidLocationEnablerModule(reactContext: ReactApplicationContext) :
 
   @ReactMethod
   fun promptForEnableLocationIfNeeded(params: ReadableMap?, promise: Promise) {
-    if (currentActivity == null) return
+    if (reactApplicationContext.currentActivity == null) return
 
     this.promise = promise
 
@@ -118,7 +118,7 @@ class AndroidLocationEnablerModule(reactContext: ReactApplicationContext) :
   }
 
   private fun isLocationProviderEnabled(): Boolean {
-    val locationManager = currentActivity?.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+    val locationManager = reactApplicationContext.currentActivity?.getSystemService(Context.LOCATION_SERVICE) as LocationManager
     if (locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
       return true
     }
@@ -126,7 +126,7 @@ class AndroidLocationEnablerModule(reactContext: ReactApplicationContext) :
     return false
   }
 
-  override fun onActivityResult(activity: Activity?, requestCode: Int, resultCode: Int, intent: Intent?) {
+  override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, intent: Intent?) {
     Log.i(TAG, "On activityResult : $requestCode")
     if (requestCode == REQUEST_CHECK_SETTINGS) {
       if (resultCode == RESULT_OK || isLocationProviderEnabled()) {
@@ -139,7 +139,7 @@ class AndroidLocationEnablerModule(reactContext: ReactApplicationContext) :
     }
   }
 
-  override fun onNewIntent(intent: Intent?) {
+  override fun onNewIntent(intent: Intent) {
     // Nothing to do
   }
 }
